ztest: add a delay between tests

nashif opened the pull request:
Some testsuites dump lots of output very fast where some systems are not
able to capture the complete output from the tests. Add a slight delay
between each test in the suite.

Fixes #50455

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ztest: add a delay between tests through CONFIG_ZTEST_TEST_DELAY_MS

Some testsuites dump lots of output very fast where some systems are not
able to capture the complete output from the tests. Add a slight delay
between each test in the suite.

New kconfig CONFIG_ZTEST_TEST_DELAY_MS is added to ztest.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
intel_adsp: when using ztest, add delay between tests

console can't handle the amount of data coming from many tests, so
introduce a delay beween testcases.

Use ZTEST_TEST_DELAY_MS to delay test execution.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

shuai1x commented:
I tested it on CAVS platform five times. And it would pass stably. So this PR will solve the issue of missing log.

fabiobaltieri commented on subsys/testsuite/ztest/Kconfig:
```diff
@@ -22,6 +22,13 @@ config ZTEST_STACK_SIZE
 	default 2048 if COVERAGE_GCOV
 	default 1024
 
+config ZTEST_TEST_DELAY
```
How about ZTEST_TEST_DELAY_MS?

fabiobaltieri commented on subsys/testsuite/ztest/src/ztest_new.c:
```diff
@@ -509,6 +509,9 @@ static int run_test(struct ztest_suite_node *suite, struct ztest_unit_test *test
 	int ret = TC_PASS;
 
+#if CONFIG_ZTEST_TEST_DELAY_MS > 0
+	k_busy_wait(CONFIG_ZTEST_TEST_DELAY_MS * 1000);
```
Sorry just noticed this as well: should probably use `USEC_PER_MSEC` instead of `1000`.

nashif committed:
tests: mem_blocks: remove board configs using dma logging

This is no longer needed and currently generating a fault with this
backend. Using the ztest delay fixes the issue and produces the complete
console output.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
